tests/readonlytest/readonly_test.go

```diff
@@ -142,6 +137,7 @@ func TestRestrictionWithConnectionPool(t *testing.T) {
 			} else {
 				done <- false
 			}
+			return
```

Comment:
That is un-common. Can we change to a pattern like:

```
if err != nil {
   if err.Error() ==
   else
   return
}
```

Reply:
Let me take a closer look.

Reply:
updated.